numba/targets/arrayobj.py

```diff
@@ -905,6 +905,44 @@ def array_argmax_impl(arry):
     return context.compile_internal(builder, array_argmax_impl, sig, args)
 
 
+@builtin
+@implement(numpy.median, types.Kind(types.Array))
+@implement("array.mean", types.Kind(types.Array))
+def array_median(context, builder, sig, args):
+    # XXX: the implementation doesn't work for the case if array is not one
+    # dimentional
+    def partition(A, p, r):
+        # TODO: make it randomized
+        x = A[r]
+        i = p-1
```

Comment:
To make things clearer, you could start at `i = p` and move `i += 1` after the swap.